File: drools-spring-v2-app/src/main/resources/static/js/core/medicine/medicine.service.js
'use strict';

angular.module('core.medicine')
	.service('MedicineService', function($http) {
		this.getAll = () => {
			return $http.get('/api/medicine');
		};
		this.getOne = (id) => {
			return $http.get('/api/medicine/'+id);
		};
		this.add = (data) => {
			return $http.post('/api/medicine/', data);
		};
		this.checkAlergies = (patientId,medicineId) => {
			return $http.get('/api/medicine/alergies/'+patientId+'/'+medicineId);
		};
		this.prescribe = (patientId,medicineId,dieaseId) => {
			return $http.get('/api/medicine/prescribe/'+patientId+'/'+ medicineId +'/'+dieaseId);
			
		};
		this.deleteMedicine = (id) =>{
			return $http.delete('/api/medicine/'+id);
		};
		
	});

File: drools-spring-v2-app/src/main/java/drools/spring/example/medicine/MedicineService.java
package drools.spring.example.medicine;

import java.util.List;

import drools.spring.example.patient.Patient;

public interface MedicineService {

	Medicine findOne(Long id);
	
	List<Medicine> findAll();
	
	Medicine add(Medicine medicine);
	
	String checkAlergies(Long patientId, Long medicineId);

	Medicine delete(Long id);
}
